aws/src/main/java/org/apache/iceberg/aws/s3/S3InputStream.java

```diff
@@ -194,9 +195,27 @@ private void openStream() throws IOException {
     }
   }
 
-  private void closeStream() throws IOException {
+  private void closeStream() {
     if (stream != null) {
-      stream.close();
+      // if we aren't at the end of the stream, and the stream is abortable, then
+      // call abort() so we don't read the remaining data
+      abortStream();
+      try {
+        stream.close();
+      } catch (Exception e) {
+        // close quietly
```

Comment:
We might want to log the underlying reason as info or warn here, just so we don't silently swallow issues that may come up.

Reply:
This will always throw on an aborted stream on a checksum check failure, so it could be very noisy

Reply:
I added a trace log